### DISCUSSION TITLE
Multiple file inputs (dynamic) on the same form

### DISCUSSION THREAD

**Discussion by merthurturk** [livewire/livewire#1102] [4. General]
I have the following form in my component. It has more than one file input, and the inputs are dynamically created based on the number of items. In order to link the uploads to the items, I use ```$uploads[$item->uuid]``` as my model (an array where keys are item uuids and values are files).

```php
<form wire:submit.prevent="save">
    @foreach ($items as $item)
        <input type="file" wire:model="uploads.{{ $item->uuid }}" />
    @endforeach

    <button type="submit">Upload</button>
</form>
```

However, there's a problem with this approach. I don't get the temporary uploaded file objects inside the model (BTW I'm using S3 driver). Here's my controller:

```php
use Livewire\Component;

class FileUploader extends Component
{
    public $uploads = [];
    
    public function save()
    {
        dd($this->uploads);
    }
}
```

... and the output:

```
array:1 [
    "051b2bbb-751e-48ba-8a65-dee437fe1271" => array:1 [
        "disk" => "do"
    ]
]
```

As you can see in the output, the file upload object is not present inside the model. So, I have two questions:

1. Is this a bug or am I pushing too much? :)
2. If this is not possible at all, how should I proceed with this? I'm looking for a way to link multiple file inputs (with single file selection each) to multiple items in the component (via an id or uuid), any suggestions?

Note: I know that S3 does not support multiple file uploads, however in this case each input is just a single file input.

**Comment by RobertCordes** [comment]
Did you manage to resolve this? I've been trying to do basically the same thing for days now. Already lost so many hours on this :-/

**Reply by shez1983** [reply]
hi - did YOU maange to solve this :p

**Reply by RobertCordes** [reply]
I can't remember :D This was so long ago.
